File: javascript/js/subscription.js
import Client from "./client";

class SubscriptionClient {
  async create() {
    const client = new Client();
    return client.create({
      resourceType: 'Subscription',
      criteria: 'Appointment?name=test',
      status: 'requested',
      channel: {
        type: 'rest-hook',
        endpoint: 'https://your-webhook/endpoint',
        header: ['Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8']
      }
    });
  }
}

export default SubscriptionClient;

File: ruby/src/subscription.rb
require './fhir_client'

class SubscriptionClient
  def create_subscription
    FhirClient.new

    subscription = FHIR::Subscription.new

    subscription.criteria = 'Appointment?name=test'
    subscription.status = 'requested'
    subscription.channel = FHIR::Subscription::Channel.new
    subscription.channel.type = 'rest-hook'
    subscription.channel.endpoint = 'https://your-webhook/endpoint'
    subscription.channel.header = ['Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8']

    FHIR::Subscription.create(subscription)
  end
end
